From fbb803061d77e67110fd75ddc65e5dac17e1ce11 Mon Sep 17 00:00:00 2001
From: Christophe Dumez <christophe.dumez@intel.com>
Date: Fri, 1 Jul 2011 15:59:43 -0700
Subject: [PATCH] MemoryCalendar::deleteIncidence() Delete non-Event
 incidences

Due to a mistake in the deleteIncidence() method implementation,
only incidences of type Event get deleted. Other types such as
Todo and Journal will return false and display a warning such as:
memorycalendar.cpp: 153 - "Journal"  not found.
---
 src/memorycalendar.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/memorycalendar.cpp b/src/memorycalendar.cpp
index 285a2e4..a34bb80 100644
--- a/src/memorycalendar.cpp
+++ b/src/memorycalendar.cpp
@@ -139,7 +139,7 @@ bool MemoryCalendar::deleteIncidence( const Incidence::Ptr &incidence )
   removeRelations( incidence );
   const Incidence::IncidenceType type = incidence->type();
   const QString uid = incidence->uid();
-  if ( d->mIncidences[Incidence::TypeEvent].remove( uid, incidence ) ) {
+  if ( d->mIncidences[type].remove( uid, incidence ) ) {
     setModified( true );
     notifyIncidenceDeleted( incidence );
     d->mDeletedIncidences[type].insert( uid, incidence );
-- 
1.7.4.4

